[zwave_proxy] Add docs (#5380)
OttoWinter/esphomedocs: content/components/zwave_proxy.md

---
description: "Instructions for setting up the Z-Wave Proxy in ESPHome."
title: "Z-Wave Proxy"
params:
  seo:
    description: Instructions for setting up the Z-Wave Proxy in ESPHome.
    image: z-wave.svg
---

The `zwave_proxy` component allows proxying of Z-Wave data frames between a
[Z-Wave Modem SoC](https://www.silabs.com/wireless/z-wave/800-series-modem-soc) and

> [Z-Wave JS](https://github.com/zwave-js) via ESPHome's {{< docref "/components/api" >}} over

Reviewer: Can we link to the specific Z-Wave JS docs on esphome?

Author: Good idea -- not sure if anything is published just yet. I'll see what I can find out 🙂

Author: Doesn't look like their docs have been updated yet. 🙈 Might have to circle back later on.

---
description: "Instructions for setting up the Z-Wave Proxy in ESPHome."
title: "Z-Wave Proxy"
params:
  seo:
    description: Instructions for setting up the Z-Wave Proxy in ESPHome.
    image: z-wave.svg
---

The `zwave_proxy` component allows proxying of Z-Wave data frames between a
[Z-Wave Modem SoC](https://www.silabs.com/wireless/z-wave/800-series-modem-soc) and

> [Z-Wave JS](https://github.com/zwave-js) via ESPHome's {{< docref "/components/api" >}} over

Author: Let's at least link to the *docs* site, though.
```suggestion
[Z-Wave JS](https://zwave-js.github.io/zwave-js/) via ESPHome's {{< docref "/components/api" >}} over
```

While ESPHome supports multiple API connections/clients, only a single client may subscribe to and receive proxied
Z-Wave data frames at any given time.

{{< /note >}}

## Configuration

```yaml
# Example configuration entry
zwave_proxy:
```

- **id** (*Optional*, [ID](#config-id)): Manually specify the ID for the `zwave_proxy`.

## Maximizing performance

Low latency is key to achieving an optimal experience with Z-Wave (or any) wireless devices.

It's important to understand that using the `zwave_proxy` *will* increase latency between your Z-Wave devices and
Z-Wave JS -- this is simply the consequence of passing messages from one medium to another.

Under near-ideal conditions:

- With a direct serial connection to the Z-Wave modem SoC, latency may be as low as approximately 20 milliseconds (ms).
- When introducing the Z-Wave proxy component, it's still possible to achieve low latency -- we've seen as low as 35
  ms! however, in practice, it's more realistic to expect 50-60 ms, assuming decent RF conditions.

In general, any duration less than 100 ms is quite acceptable in terms of latency; this value is generally a good
target to keep in mind.

## Maximizing reliability

> In addition to low latency, reliability is also critical to an optimal experience.

Reviewer: I'd suggest simplifying this to a bottom line up front approach that explains that if reliability is important to use Ethernet

```diff
diff --git a/content/components/zwave_proxy.md b/content/components/zwave_proxy.md
index 83b6937a..f3d654d8 100644
--- a/content/components/zwave_proxy.md
+++ b/content/components/zwave_proxy.md
@@ -56,15 +56,14 @@ target to keep in mind.
 
 In addition to low latency, reliability is also critical to an optimal experience.
 
-- While most convenient, Wi-Fi is subject to interference from other, nearby RF devices and environmental conditions.
-  These factors can (and do) periodically cause (momentary) instability of the signal. In many cases, Wi-Fi devices
-  recover automatically, but this comes at the expense of degraded performance (latency) and, in more extreme cases,
-  brief loss of connectivity. This is not desirable for applications such as the Z-Wave proxy and it's important to
-  keep in mind as you determine how you'll connect it.
-- A wired Ethernet connection will, in all cases, prove more reliable than a Wi-Fi connection for the Z-Wave proxy
-  application. It's much more stable and predictable in terms of latency and the connection will not arbitrarily drop
-  (unless, of course, you unplug the cable). For this reason, we highly recommend using a wired connection to achieve
-  the best experience.
+**A wired Ethernet connection will provide the best reliability for the Z-Wave proxy.** Ethernet offers:
+- More stable and predictable latency
+- No connection drops from RF interference
+- Consistent performance regardless of environmental conditions
+
+### Wi-Fi considerations
+
+Wi-Fi is subject to interference from other RF devices and environmental conditions. These factors can periodically cause momentary signal instability. While Wi-Fi devices typically recover automatically, this comes at the expense of temporarily degraded performance (latency) and, in more extreme cases, brief loss of connectivity.
 
 If you choose to use Wi-Fi to bridge your Z-Wave modem to Z-Wave JS:

```

> In addition to low latency, reliability is also critical to an optimal experience.
> 
> - While most convenient, Wi-Fi is subject to interference from other, nearby RF devices and environmental conditions.
>   These factors can (and do) periodically cause (momentary) instability of the signal. In many cases, Wi-Fi devices
>   recover automatically, but this comes at the expense of degraded performance (latency) and, in more extreme cases,
>   brief loss of connectivity. This is not desirable for applications such as the Z-Wave proxy and it's important to
>   keep in mind as you determine how you'll connect it.
> - A wired Ethernet connection will, in all cases, prove more reliable than a Wi-Fi connection for the Z-Wave proxy
>   application. It's much more stable and predictable in terms of latency and the connection will not arbitrarily drop
>   (unless, of course, you unplug the cable). For this reason, we highly recommend using a wired connection to achieve
>   the best experience.
> 

Reviewer: ```suggestion
In addition to low latency, reliability is also critical to an optimal experience.

**A wired Ethernet connection will provide the best reliability for the Z-Wave proxy.** Ethernet offers:
- More stable and predictable latency
- No connection drops from RF interference
- Consistent performance regardless of environmental conditions

### Wi-Fi considerations

Wi-Fi is subject to interference from other RF devices and environmental conditions. These factors can periodically cause momentary signal instability. While Wi-Fi devices typically recover automatically, this comes at the expense of temporarily degraded performance (latency) and, in more extreme cases, brief loss of connectivity.

```

It's important to understand that using the `zwave_proxy` *will* increase latency between your Z-Wave devices and
Z-Wave JS -- this is simply the consequence of passing messages from one medium to another.

Under near-ideal conditions:

- With a direct serial connection to the Z-Wave modem SoC, latency may be as low as approximately 20 milliseconds (ms).
- When introducing the Z-Wave proxy component, it's still possible to achieve low latency -- we've seen as low as 35
  ms! however, in practice, it's more realistic to expect 50-60 ms, assuming decent RF conditions.

In general, any duration less than 100 ms is quite acceptable in terms of latency; this value is generally a good
target to keep in mind.

## Maximizing reliability

In addition to low latency, reliability is also critical to an optimal experience.

**A wired Ethernet connection will provide the best reliability for the Z-Wave proxy.** Ethernet offers:

- More stable and predictable latency
- No connection drops or lag due to RF interference
- Immunity to environmental conditions

### Wi-Fi considerations

Wi-Fi is impacted by:

> - Interference from other RF devices

Author: ```suggestion

- Interference from other RF devices
```

In addition to low latency, reliability is also critical to an optimal experience.

**A wired Ethernet connection will provide the best reliability for the Z-Wave proxy.** Ethernet offers:

> - More stable and predictable latency

Reviewer: ```suggestion

- More stable and predictable latency
```
